src/librustc_typeck/astconv.rs

```diff
@@ -1345,9 +1344,11 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
 
         let (assoc_ident, def_scope) =
             tcx.adjust_ident_and_get_scope(binding.item_name, candidate.def_id(), hir_ref_id);
+
+        // FIXME(ecstaticmorse): Can this use `find_by_name_and_kind` instead?
```

Comment:
Yes, but that will do the adjustments calculations (which we already did in `adjust_ident_and_get_scope` above) for the second time.
If perf run says it's ok, then it can be done (here and next to other uses of `adjust_ident_and_get_scope`).